cmake/SetupCompilers.cmake

```diff
@@ -59,6 +59,9 @@ if (CMAKE_CXX_COMPILER_ID MATCHES GNU)
 elseif (CMAKE_CXX_COMPILER_ID MATCHES AppleClang)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 elseif (CMAKE_CXX_COMPILER_ID MATCHES Clang)
+  set(CMAKE_CXX_STANDARD 11) # C++11...
+  set(CMAKE_CXX_STANDARD_REQUIRED ON) #...is required...
+  set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
>  it was adding in -std=gnu++11 in addition to -std=c++11

I was just wondering about this -- good to know that CMake doesn't do the right thing. In this case, we should probably do it the CMake Way ™. The only problem is some compilers don't have configurations for CMake or CMake doesn't know how to feature test them. We'll probably need to vet that out in our Cmake helpers.